third-party/realdds/src/dds-device-broadcaster.cpp

```diff
@@ -195,105 +182,43 @@ bool dds_device_broadcaster::add_dds_device( const device_info & dev_info )
 
 bool dds_device_broadcaster::create_device_writer( const device_info & dev_info )
 {
-    // Create a data writer for the topic
-    DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
-    wqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
-    wqos.durability().kind = VOLATILE_DURABILITY_QOS;
-
-    //---------------------------------------------------------------------------------------
-    // The writer-reader handshake is done on UDP, even when data_sharing (shared memory) is used for
-    // actual messaging. This means it is possible to send a message and receive it on the reader�s
-    // side even before the UDP handshake is complete:
-    //   1. The writer goes up and broadcasts its presence periodically; no readers exist
-    //   2. The reader joins and broadcasts its presence, again periodically; it doesn�t know about the writer yet
-    //   3. The writer sees the reader (in-between broadcasts) so sends a message
-    //   4. The reader gets the message and discards it because it does not yet recognize the writer
-    // This depends on timing. When shared memory is on, step 3 is so fast that this miscommunication is much more likely.
-    // This is a known gap in the DDS standard.
-    //
-    // We can either insert a sleep between writer creation and message sending or we can disable
-    // data_sharing for this topic, which we did here.
-    // (See https://github.com/eProsima/Fast-DDS/issues/2641)
+    if( ! _topic )
+        _topic = topics::flexible_msg::create_topic( _publisher->get_participant(), "realsense/device-info" );
 
-    // wqos.data_sharing().automatic();
-    wqos.data_sharing().off();
-    //---------------------------------------------------------------------------------------
+    // Create a data writer for the topic
+    auto writer = std::make_shared< dds_client_listener >( _topic, _publisher, this );
+    if( ! writer )
+        return false;
 
-    std::shared_ptr< dds_client_listener > writer_listener = std::make_shared< dds_client_listener >( this );
+    dds_topic_writer::qos wqos( RELIABLE_RELIABILITY_QOS ); //RELIABLE_RELIABILITY_QOS default but worth mentioning
+    writer->run( wqos );
 
-    auto it_inserted = _device_handle_by_sn.emplace(
-        dev_info.serial,
-        dds_device_handle{ dev_info,
-                           _publisher->create_datawriter( _topic, wqos, writer_listener.get() ),
-                           writer_listener } );
```

Comment:
Please check with this change that all permutations work 
(Or make sure we have a test for it)

server down --> client up --> server up
client down --> server up --> client up

Reply:
I don't think that client up before server worked even before this change. I have a task in my todo list to check and fix this issue from before.
Will verify and update here...

Reply:
As discussed , will be handled on next PR